### PR TITLE
[TIMOB-19200] Android: implement 'noresults' event on ListView

### PR DESCRIPTION
**Pull request by ashcoding**
Jira: https://jira.appcelerator.org/browse/TIMOB-19200

Test code for listview:

``` javascript
var win = Ti.UI.createWindow({backgroundColor: 'grey', fullscreen: true});
var search = Titanium.UI.createSearchBar({
 barColor:'#000', 
 showCancel:true,
 height:43,
 top:0,
});
search.addEventListener('cancel', function(){
    search.blur();
});
// for textSearch, use the change event to update the search value
// search.addEventListener('change', function(e){
//     listView.searchText = e.value;
// });
var listView = Ti.UI.createListView({searchView: search, caseInsensitiveSearch: true});
// for textSearch, add the search bar or text field as a header view
// var listView = Ti.UI.createListView({headerView: search, caseInsensitiveSearch: true});
var listSection = Ti.UI.createListSection();
var fruits = ['Papaya', 'Peach', 'Pear', 'Persimmon', 'Pineapple', 'Pluot', 'Pomegranate'];
var data = [];
for (var i = 0; i < fruits.length; i++) {
data.push({
    properties: {title: fruits[i], searchableText: fruits[i]}
});
}
listSection.setItems(data);
listView.addEventListener('noresults', function(e){
    alert("No results found!");
});
listView.sections = [listSection];
win.add(listView);
win.open();
```

Test code for tableview:

``` javascript
var win = Ti.UI.createWindow({
    backgroundColor: 'blue',
    fullscreen: false
});

// Use action bar search view
var search = Ti.UI.Android.createSearchView({
    hintText: "Table Search"
});

win.activity.onCreateOptionsMenu = function(e) {
    var menu = e.menu;
    var menuItem = menu.add({
        title: 'Table Search',
        actionView : search,
        icon: (Ti.Android.R.drawable.ic_menu_search ? Ti.Android.R.drawable.ic_menu_search : "my_search.png"),
        showAsAction: Ti.Android.SHOW_AS_ACTION_IF_ROOM | Ti.Android.SHOW_AS_ACTION_COLLAPSE_ACTION_VIEW
    });
};

var data = [];
data.push(Ti.UI.createTableViewRow({title:'Apple'}));
data.push(Ti.UI.createTableViewRow({title:'Banana'}));
data.push(Ti.UI.createTableViewRow({title:'Orange'}));
data.push(Ti.UI.createTableViewRow({title:'Raspberry'}));

var tableview = Titanium.UI.createTableView({
    data: data,
    search: search,
    searchAsChild: false
});

tableview.addEventListener('noresults', function(e){
    alert("No results found!");
});

win.add(tableview);
win.open();
```

### CODE REVIEW

File: apidoc/Titanium/UI/ListView.yml [hunk 1]
```diff
@@ -356,8 +356,11 @@ events:
 
         While the `searchView` API automatically shows a **No Results** cell in the list view, the `searchText` API does not do so.
         Developers can listen to this event and perform appropriate actions when the search result set is empty.
-    platforms: [iphone, ipad]
-    since: "3.3.0"
+    platforms: [android, iphone, ipad]
+    since:
+        android: 4.1.0
```
Comment (hieupham007) [inline apidoc/Titanium/UI/ListView.yml]: 4.2.0

File: android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java [hunk 2]
```diff
@@ -650,12 +650,16 @@ private void setHeaderOrFooterView (Object viewObj, boolean isHeader) {
...
+		int numResults = 0;
 		if (searchText != null) {
 			for (int i = 0; i < sections.size(); ++i) {
 				ListSectionProxy section = sections.get(i);
-				section.applyFilter(searchText);
+				numResults += section.applyFilter(searchText);
 			}
 		}
+		if (numResults == 0) {
+			fireEvent("noresults", null);
```
Comment (hieupham007) [inline android/modules/ui/src/java/ti/modules/titanium/ui/widget/listview/TiListView.java]: Please make a TiC constant for this.

**Comment by hieupham007** [comment]
Code reviewed and functionally tested. Looks good, just a minor comment

**Comment by ashcoding** [comment]
@hieupham007 addressed comments.

**Comment by hieupham007** [comment]
Hmm, why are you overwriting the commit? It's better to have multiple commits so we can track what is changed. Next time please don't overwrite the commit with new changes.

**Comment by ashcoding** [comment]
Will do.

**Pull request merged**